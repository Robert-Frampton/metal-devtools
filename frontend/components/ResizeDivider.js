import Component, {Config} from 'metal-jsx';
import {bindAll} from 'lodash';

/**
 * Component used emit an event when dragged horizontally
 */
class ResizeDivider extends Component {
  created() {
    bindAll(this, 'doDrag', 'initDrag', 'stopDrag');
  }

  attached() {
    this.on('mousedown', this.initDrag);
  }

  doDrag({clientX}) {
    this.props.onResize({clientX});
  }

  initDrag() {
    document.body.addEventListener('mousemove', this.doDrag, false);
    document.body.addEventListener('mouseup', this.stopDrag, false);
  }

  stopDrag() {
    document.body.removeEventListener('mousemove', this.doDrag, false);
    document.body.removeEventListener('mouseup', this.stopDrag, false);
  }

  render() {
    return <div class="resize-container" />;
  }
}

ResizeDivider.PROPS = {
  onResize: Config.func().required()
};

export default ResizeDivider;
