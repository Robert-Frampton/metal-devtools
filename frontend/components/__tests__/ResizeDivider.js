jest.unmock('../ResizeDivider');

import ResizeDivider from '../ResizeDivider';

describe('ResizeDivider', () => {
	it('should render', () => {
		const component = new ResizeDivider({onResize: jest.fn()});

		expect(snap(component)).toMatchSnapshot();
	});
});
