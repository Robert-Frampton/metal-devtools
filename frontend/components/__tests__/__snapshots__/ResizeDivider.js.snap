exports[`ResizeDivider should render 1`] = `undefined`;
